### PR TITLE
Remove unneeded decoding from AuthTokenProcessorHandler

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Fixes bug where the SAMLAuthenticator expects the provided key to be base64 encoded otherwise it would cause saml-authenticator initialization to fail.

Front-end tests pass exchange-key in URL encoded format. Another solution to this would be to remove `getDecoder()` completely. This works and I've tested it locally as well.

I proceed with the approach to remove decoding as that was the state prior to library change.

The tests didn't catch this because they use `abc` as exchange_key. I've updated it to reflect the exchange key used in front-end tests: `6aff3042-1327-4f3d-82f0-40a157ac4464`. 

Backport of this PR will unblock frontend CI

### Issues Resolved

- Resolves: https://github.com/opensearch-project/security/issues/3604

### Testing
Existing tests.

### Check List
- [x] New functionality includes testing
~- [ ] New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java [hunk 1]
```diff
@@ -99,21 +98,17 @@ class AuthTokenProcessorHandler {
...
 
-        if (samlSubjectKey == null || samlSubjectKey.length() == 0) {
+        if (samlSubjectKey == null || samlSubjectKey.isEmpty()) {
             // If subjectKey == null, get subject from the NameID element.
             // Thus, this is a valid configuration.
             samlSubjectKey = null;
         }
 
-        if (samlRolesSeparator == null || samlRolesSeparator.length() == 0) {
-            samlRolesSeparator = null;
```
Comment (DarshitChanpura) [inline src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:114]: this assignment was unused.

File: src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java [hunk 2]
```diff
@@ -128,12 +123,7 @@ Optional<SecurityResponse> handle(RestRequest restRequest) throws Exception {
                 sm.checkPermission(new SpecialPermission());
             }
 
-            return AccessController.doPrivileged(new PrivilegedExceptionAction<Optional<SecurityResponse>>() {
-                @Override
-                public Optional<SecurityResponse> run() throws SamlConfigException, IOException {
-                    return handleLowLevel(restRequest);
-                }
-            });
+            return AccessController.doPrivileged((PrivilegedExceptionAction<Optional<SecurityResponse>>) () -> handleLowLevel(restRequest));
```
Comment (DarshitChanpura) [inline src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:126]: seems like a quick code cleanup and so included in this PR. Same pattern can be found in other files but that is out of scope of this PR.

File: src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java [hunk 3]
```diff
@@ -171,7 +171,7 @@ public void basicTest() throws Exception {
             .put(IDP_METADATA_URL, mockSamlIdpServer.getMetadataUri())
             .put("kibana_url", "http://wherever")
             .put("idp.entity_id", mockSamlIdpServer.getIdpEntityId())
-            .put("exchange_key", "abc")
+            .put("exchange_key", "6aff3042-1327-4f3d-82f0-40a157ac4464")
```
Comment (RyanL1997) [inline src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java:174]: why we need this btw?

Comment (DarshitChanpura) [reply src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java:174]: The tests didn't catch the error because they use `abc` as exchange_key. I've updated it to reflect the exchange key used in front-end tests: `6aff3042-1327-4f3d-82f0-40a157ac4464`

File: src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java [hunk 4]
```diff
@@ -252,7 +242,7 @@ JWK createJwkFromSettings(Settings settings, Settings jwtSettings) throws Except
         String exchangeKey = settings.get("exchange_key");
 
         if (!Strings.isNullOrEmpty(exchangeKey)) {
```
Comment (cwperks) [inline src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:244]: Before the JWT library change, the CXF library assumed that the exchangeKey was Base64 encoded:

1. https://github.com/opensearch-project/security/blob/2.11/src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java#L257
2. https://github.com/apache/cxf/blob/main/rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/jws/JwsUtils.java#L109-L112
3. https://github.com/apache/cxf/blob/main/rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/common/JoseUtils.java#L112-L114
4. https://github.com/apache/cxf/blob/main/rt/security/src/main/java/org/apache/cxf/rt/security/crypto/CryptoUtils.java#L669-L675

With the change to nimbus, the configured value should also be assumed to be base64 encoded. The nimbus call:

```
new OctetSequenceKey.Builder(exchangeKey.getBytes(StandardCharsets.UTF_8))
```

will perform Base64 encoded as part of the [constructor of the builder](https://github.com/felx/nimbus-jose-jwt/blob/master/src/main/java/com/nimbusds/jose/jwk/OctetSequenceKey.java#L179-L186) so the code above decodes the configured value, pads it (for backwards compatibility?) and then encodes again. 

The code to remove the Base64 decoding here would be needed for backwards compatibility so it cannot be removed.

Comment (DarshitChanpura) [reply src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:244]: > The code to remove the Base64 decoding here would be needed for backwards compatibility so it cannot be removed.

I'm slightly confused. Are you suggesting we move forward with getUrlDecoder in place of getDecoder?

Comment (cwperks) [reply src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:244]: Yes, `getUrlDecoder()` would be in line with the previous JWT functionality using cxf.

Comment (DarshitChanpura) [reply src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java:245]: I've also replaced with url decoder to keep it backwards compatible.

File: src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java (hunk 3 above)
Comment (cwperks) [inline src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java:174]: Can there be 2 separate tests to confirm key behavior in both scenarios?

Comment (DarshitChanpura) [reply src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java]: done

**Review by cwperks — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]
